docs/Validation.md

```diff
@@ -362,52 +362,88 @@ const CustomerCreate = () => (
 
 ## Server-Side Validation
 
-You can use the errors returned by the dataProvider mutation as a source for the validation. In order to display the validation errors, a custom `save` function needs to be used:
+Server-side validation is supported out of the box. It requires that the dataProvider throws an error with the following shape:
 
-{% raw %}
-```jsx
-import * as React from 'react';
-import { useCallback } from 'react';
-import { Create, SimpleForm, TextInput, useCreate, useRedirect, useNotify } from 'react-admin';
-
-export const UserCreate = () => {
-    const redirect = useRedirect();
-    const notify = useNotify();
-
-    const [create] = useCreate();
-    const save = useCallback(
-        async values => {
-            try {
-                await create(
-                    'users',
-                    { data: values },
-                    { returnPromise: true }
-                );
-                notify('ra.notification.created', {
-                    type: 'info',
-                    messageArgs: { smart_count: 1 },
-                });
-                redirect('list');
-            } catch (error) {
-                if (error.body.errors) {
-                    // The shape of the returned validation errors must match the shape of the form
-                    return error.body.errors;
+```
+{
+    body: {
+        errors: {
+            title: 'An article with this title already exists. The title must be unique.',
+            date: 'The date is required',
+            tags: { message: "The tag 'agrriculture' doesn't exist" },
+        }
+    }
+}
+```
+
+**Tip**: The shape of the returned validation errors must match the form shape: each key needs to match a `source` prop.
+
+**Tip**: The returned validation errors might have any validation format we support (simple strings or object with message and args) for each key.
```

Comment:
This is, according to my tests, not entirely true. Translation strings are supported, either returned directly or as the `message` prop, but translation **args** do not seem to work.

Anyway, fixing this would fall beyond the scope of this PR IMO, so I'll just remove this part from the docs.